pkg/cmd/cli/cmd/idle.go

```diff
@@ -0,0 +1,557 @@
+}
+
+type scaleInfo struct {
+	scale *extensions.Scale
+	obj   runtime.Object
+}
+
+// RunIdle runs the idling command logic, taking a list of resources or services in a file, scaling the associated
+// scalables to zero, and annotating the associated endpoints objects with the scalables to unidle when they receive traffic.
+func RunIdle(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, filename string, loadByService, dryRun bool) error {
+	namespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	nowTime := time.Now().UTC()
+
+	// figure out which endpoints and resources we need to idle
+	byService, byScalable, err := calculateIdlableAnnotationsByService(f, namespace, filename, out)
+
+	if err != nil {
+		if len(byService) == 0 || len(byScalable) == 0 {
+			return fmt.Errorf("no valid scalables found to idle: %v", err)
+		}
+		fmt.Fprintf(out, "Errors while finding scalable resources to idle: %v\nContinuing for valid scalables...\n", err)
+	}
+
+	oclient, kclient, err := f.Clients()
+	if err != nil {
+		return err
+	}
+	delegScaleGetter := osclient.NewDelegatingScaleNamespacer(oclient, kclient).Scales(namespace)
+	dcGetter := deployclient.New(oclient.RESTClient).DeploymentConfigs(namespace)
+	rcGetter := clientset.FromUnversionedClient(kclient).ReplicationControllers(namespace)
+
+	replicas := make(map[unidlingapi.CrossGroupObjectReference]int32, len(byScalable))
+	toScale := make(map[unidlingapi.CrossGroupObjectReference]scaleInfo)
+
+	mapper, typer := f.Object(false)
+
+	scaleAnnotater := utilunidling.NewScaleAnnotater(delegScaleGetter, dcGetter, rcGetter, func(ref unidlingapi.CrossGroupObjectReference, annotations map[string]string) {
+		annotations[unidlingapi.IdledAtAnnotation] = nowTime.UTC().Format(time.RFC3339)
+	})
+
+	// first, collect the scale info
+	for scaleRef, svcName := range byScalable {
+		obj, scale, err := scaleAnnotater.GetObjectWithScale(scaleRef)
+		if err != nil {
+			fmt.Fprintf(out, "Unable to get scale for %s %s/%s, not marking that scalable as idled...\n", scaleRef.Kind, namespace, scaleRef.Name)
+			svcInfo := byService[svcName]
+			delete(svcInfo.scalables, scaleRef)
+			continue
+		}
+		replicas[scaleRef] = scale.Spec.Replicas
+		toScale[scaleRef] = scaleInfo{scale: scale, obj: obj}
+	}
+
+	// annotate the endpoints objects to indicate which scalables need to be unidled on traffic
+	for serviceName, info := range byService {
+		if info.obj.Annotations == nil {
+			info.obj.Annotations = make(map[string]string)
+		}
+		refsWithScale, err := pairScalesWithIdlables(serviceName, info.obj.Annotations, info.scalables, replicas)
+		if err != nil {
+			fmt.Fprintf(out, "Unable to mark service %s/%s as idled: %v", namespace, serviceName, err)
+			continue
+		}
+
+		if !dryRun {
+			if len(info.scalables) == 0 {
+				fmt.Fprintf(out, "No scalables marked as idled for service %s/%s, not marking as idled...\n", namespace, serviceName)
```

Comment:
You need to follow our consistent output scheme - all errors are prefixed with `error: ` or `warning: ` and are printed to stdout.  If `-o name` is printed you need to be printing `resource/name` (we can defer that for now potentially, but leave a todo).  All other output goes to stdout.